test/e2e/apps/daemon_restart.go

```diff
@@ -60,53 +60,68 @@ const (
 // restartDaemonConfig is a config to restart a running daemon on a node, and wait till
 // it comes back up. It uses ssh to send a SIGTERM to the daemon.
 type restartDaemonConfig struct {
-	nodeName     string
-	daemonName   string
-	healthzPort  int
-	pollInterval time.Duration
-	pollTimeout  time.Duration
+	nodeName            string
+	daemonName          string
+	insecureHealthzPort int
+	secureHealthzPort   int
+	pollInterval        time.Duration
+	pollTimeout         time.Duration
 }
 
 // NewRestartConfig creates a restartDaemonConfig for the given node and daemon.
-func NewRestartConfig(nodeName, daemonName string, healthzPort int, pollInterval, pollTimeout time.Duration) *restartDaemonConfig {
+func NewRestartConfig(nodeName, daemonName string, insecureHealthzPort int, secureHealthzPort int, pollInterval, pollTimeout time.Duration) *restartDaemonConfig {
 	if !framework.ProviderIs("gce") {
 		framework.Logf("WARNING: SSH through the restart config might not work on %s", framework.TestContext.Provider)
 	}
 	return &restartDaemonConfig{
-		nodeName:     nodeName,
-		daemonName:   daemonName,
-		healthzPort:  healthzPort,
-		pollInterval: pollInterval,
-		pollTimeout:  pollTimeout,
+		nodeName:            nodeName,
+		daemonName:          daemonName,
+		insecureHealthzPort: insecureHealthzPort,
+		secureHealthzPort:   secureHealthzPort,
+		pollInterval:        pollInterval,
+		pollTimeout:         pollTimeout,
 	}
 }
 
 func (r *restartDaemonConfig) String() string {
 	return fmt.Sprintf("Daemon %v on node %v", r.daemonName, r.nodeName)
 }
 
-// waitUp polls healthz of the daemon till it returns "ok" or the polling hits the pollTimeout
+// waitUp polls healthz of the daemon through secure or insecure port till it returns "ok" or the polling
+// hits the pollTimeout
 func (r *restartDaemonConfig) waitUp() {
 	framework.Logf("Checking if %v is up by polling for a 200 on its /healthz endpoint", r)
-	healthzCheck := fmt.Sprintf(
-		"curl -s -o /dev/null -I -w \"%%{http_code}\" http://localhost:%v/healthz", r.healthzPort)
+	var healthzChecks []string
```

Comment:
A nit here: is it possible to have both http and https at the same time? If not, maybe change to `var healthzCheck string` to keep code simpler. And also L100 would be `else if`.